### PR TITLE
Add a setting to force LPAR to set defaults in NVRAM before booting

### PR DESCRIPTION
**Pull request by alvarocarvajald**
This commit introduces the `SET_NVRAM_DEFAULTS` setting, that will cause the functions in lib/bootloader_pvm.pm to boot the LPAR into the Open Firmware command line prompt and issue a `SET_NVRAM_DEFAULTS` to reset the values in the LPAR's NVRAM to its defaults. This can help to fix some boot issues caused by aborted boot processes in the LPAR, which in turn prevent the GRUB menu from being loaded. If set, LPAR will boot into the Open Firmware prompt, issue the command and then continue to boot normally. If not set, it will boot as usual.

[type description here, PLEASE, REMOVE THIS LINE, PLACEHOLDER, BEFORE SUBMITTING THIS PULL REQUEST]

- Related ticket: N/A
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/-/merge_requests/1664
- Verification run: http://mango.qe.nue2.suse.org/tests/overview?build=88.1&distri=sle&version=15-SP6&groupid=1

### CODE REVIEW

File: lib/bootloader_pvm.pm [hunk 1]
```diff
@@ -170,6 +194,34 @@ sub boot_pvm {
...
+
+Check if an LPAR identified by C<$lpar_id> in the Power machine C<$hmc_machine_name> is down, by querying with the C<lssyscfg> command.
+Command will check 24 times in a loop while waiting 5 seconds between each run, and exit as soon as the LPAR is down. Check is performed
+by a needle match looking for the B<LPAR IS DOWN> text which is printed by the check script on success.
+
+=cut
+
+sub check_lpar_is_down {
+    my ($hmc_machine_name, $lpar_id) = @_;
+    enter_cmd("for ((i=0\; i<24\; i++)); do lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state | grep -q 'Not Activated' && echo 'LPAR IS DOWN' && break || echo 'Waiting for lpar $lpar_id to shutdown' && sleep 5 ; done ");
```
Comment (mpagot) [inline lib/bootloader_pvm.pm:209]: why the for is there and not in perl?

File: lib/bootloader_pvm.pm [hunk 2]
```diff
@@ -190,8 +242,24 @@ sub boot_hmc_pvm {
...
     enter_cmd("chsysstate -r lpar -m $hmc_machine_name -o shutdown --immed --id $lpar_id ");
-    enter_cmd("for ((i=0\; i<24\; i++)); do lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state | grep -q 'Not Activated' && echo 'LPAR IS DOWN' && break || echo 'Waiting for lpar $lpar_id to shutdown' && sleep 5 ; done ");
-    assert_screen 'lpar-is-down', 120;
+    check_lpar_is_down($hmc_machine_name, $lpar_id);
+
+    # Restore LPAR's NVRAM defaults if SET_NVRAM_DEFAULTS setting is present
+    if (get_var('SET_NVRAM_DEFAULTS')) {
+        # Boot into open firmware (of) first to issue a SET_NVRAM_DEFAULTS command
+        enter_cmd("chsysstate -r lpar -m $hmc_machine_name -o on -b of --id $lpar_id ");
+        enter_cmd "mkvterm -m $hmc_machine_name --id $lpar_id";
```
Comment (mpagot) [inline lib/bootloader_pvm.pm:251]:
```suggestion
        enter_cmd("mkvterm -m $hmc_machine_name --id $lpar_id");
```

Just to be as all the other in this PR

**Review by mpagot — APPROVED** [review]
LGTM, with some comments and questions

File: lib/bootloader_pvm.pm [hunk 3]
```diff
@@ -170,6 +194,34 @@ sub boot_pvm {
     }
 }
 
+=head2 check_lpar_is_down
+
+ check_lpar_is_down($hmc_machine_name, $lpar_id);
+
+Check if an LPAR identified by C<$lpar_id> in the Power machine C<$hmc_machine_name> is down, by querying with the C<lssyscfg> command.
+Command will check 24 times in a loop while waiting 5 seconds between each run, and exit as soon as the LPAR is down. Check is performed
```
[the remark this answers is not in the document]
Comment (alvarocarvajald) [reply lib/bootloader_pvm.pm:202]: Stop looping.

File: lib/bootloader_pvm.pm (hunk 1 above)
Comment (alvarocarvajald) [reply lib/bootloader_pvm.pm:209]: It was there already. I just moved it to a function as the changes I'm introducing require to call this piece of code twice and didn't want to repeat myself.

File: lib/bootloader_pvm.pm (hunk 2 above)
Comment (alvarocarvajald) [reply lib/bootloader_pvm.pm]: Done.

**Comment by alvarocarvajald** [comment]
VR after latest changes: http://mango.qe.nue2.suse.org/tests/5820

File: lib/bootloader_pvm.pm (hunk 1 above)
Comment (okurz) [inline lib/bootloader_pvm.pm:209]:
```suggestion
    enter_cmd("for i in {0..24}; do lssyscfg -m $hmc_machine_name -r lpar --filter \"\"lpar_ids=$lpar_id\"\" -F state | grep -q 'Not Activated' && echo 'LPAR IS DOWN' && break || echo 'Waiting for lpar $lpar_id to shutdown' && sleep 5 ; done ");
```

**Review by okurz — COMMENTED** [review]
@nicksinger

File: lib/bootloader_pvm.pm (hunk 1 above)
Comment (alvarocarvajald) [reply lib/bootloader_pvm.pm]: Thanks. Done.

**Pull request merged**